src/plot_api/plot_config.js

```diff
@@ -14,212 +14,425 @@
+            'the resulting graphs.'
+        ].join(' ')
+    },
+
+    displayModeBar: {
+        valType: 'enumerated',
+        values: ['hover', true, false],
+        dflt: 'hover',
+        description: [
+            'Determines the mode bar display mode.',
+            'If *true*, the mode bar is always visible.',
+            'If *false*, the mode bar is always hidden.',
+            'If *hover*, the mode bar is visible while the mouse cursor',
+            'is on the graph container.'
+        ].join(' ')
+    },
+    showSendToCloud: {
+        valType: 'boolean',
+        dflt: false,
+        description: [
+            'Should we include a ModeBar button, labeled "Edit in Chart Studio",',
+            'that sends this chart to plot.ly or another plotly server as specified',
+            'by `plotlyServerURL` for editing, export, etc? Prior to version 1.43.0',
+            'this button was included by default, now it is opt-in using this flag.',
+            'Note that this button can (depending on `plotlyServerURL`) send your data',
+            'to an external server. However that server does not persist your data',
+            'until you arrive at the Chart Studio and explicitly click "Save".'
+        ].join(' ')
+    },
+    modeBarButtonsToRemove: {
+        valType: 'any',
+        dflt: [],
+        description: [
+            'Remove mode bar buttons by name.',
+            'See ./components/modebar/buttons.js for the list of names.'
+        ].join(' ')
+    },
+    modeBarButtonsToAdd: {
+        valType: 'any',
+        dflt: [],
+        description: [
+            'Add mode bar button using config objects',
+            'See ./components/modebar/buttons.js for list of arguments.'
+        ].join(' ')
+    },
+    modeBarButtons: {
+        valType: 'any',
+        dflt: false,
+        description: [
+            'Define fully custom mode bar buttons as nested array,',
+            'where the outer arrays represents button groups, and',
+            'the inner arrays have buttons config objects or names of default buttons',
+            'See ./components/modebar/buttons.js for more info.'
+        ].join(' ')
+    },
+    toImageButtonOptions: {
+        valType: 'any',
+        dflt: {},
+        description: [
+            'Statically override options for toImage modebar button',
+            'allowed keys are format, filename, width, height, scale',
+            'see ../components/modebar/buttons.js'
+        ].join(' ')
+    },
+    displaylogo: {
+        valType: 'boolean',
+        dflt: true,
+        description: [
+            'Determines whether or not the plotly logo is displayed',
+            'on the end of the mode bar.'
+        ].join(' ')
+    },
+    watermark: {
+        valType: 'boolean',
+        dflt: false,
+        description: 'watermark the images with the company\'s logo'
+    },
+
+    plotGlPixelRatio: {
+        valType: 'number',
+        dflt: 2,
+        min: 0,
```

Comment:
Not sure about the minimum. Could we use '1' instead of '0'?

Reply:
Good call! Done https://github.com/plotly/plotly.js/pull/3376/commits/bed30cd0b242a8552d6d29f57a6a6bc206f16b2c